File: API/DTOs/Statistics/UserReadStatistics.cs
using System;
using System.Collections.Generic;

namespace API.DTOs.Statistics;
#nullable enable

public sealed record UserReadStatistics
{
    /// <summary>
    /// Total number of pages read
    /// </summary>
    public long TotalPagesRead { get; set; }
    /// <summary>
    /// Total number of words read
    /// </summary>
    public long TotalWordsRead { get; set; }
    /// <summary>
    /// Total time spent reading
    /// </summary>
    public long TimeSpentReading { get; set; }
    /// <summary>
    /// Last time user read anything
    /// </summary>
    public DateTime? LastActiveUtc { get; set; }
    public double AvgHoursPerWeekSpentReading { get; set; }

}

File: UI/Web/src/app/statistics/_models/user-read-statistics.ts
import { StatCount } from "./stat-count";

export interface UserReadStatistics {
    totalPagesRead: number;
    totalWordsRead: number;
    timeSpentReading: number;
    chaptersRead: number;
    lastActive: string;
    avgHoursPerWeekSpentReading: number;
    percentReadPerLibrary: Array<StatCount<number>>;
}